ext/pdo/pdo_dbh.c

```diff
@@ -486,6 +487,24 @@ static void pdo_stmt_construct(zend_execute_data *execute_data, pdo_stmt_t *stmt
 }
 /* }}} */
 
+static void pdo_stmt_set_select_all(pdo_stmt_t *stmt) /* {{{ */
+{
+	/* Avoid modifying stmt->query_string */
+	zend_string *statement = zend_string_init(ZSTR_VAL(stmt->query_string), ZSTR_LEN(stmt->query_string), 0);
+	char *statement_val = ZSTR_VAL(statement);
+	size_t statement_len = ZSTR_LEN(statement);
+	char *end = statement_val + statement_len;
+	php_strtolower(statement_val, statement_len);
+
+	stmt->select_all = 0;
+	if (php_memnstr(statement_val, "select", sizeof("select") - 1, end) && php_memnstr(statement_val, "*", sizeof("*") - 1, end)) {
```

Comment:
This might not be the best idea. What if we always reset the columns (regardless of the query)? Would that be a real performance issue?

Reply:
Thanks for your comment.
 I'm sorry to reply you so late.
My English is poor but I will try my best to express my thought clearly.

When sql is just like
```sql
select * from test where id = ?
```
It will reset the columns in order to get the latest columns information.

When sql is just like
```sql
select foo from test where id = ?
```
it will not reset columns.

Reply:
stmt->columns will not change after the first execute, the key of result set will not change too although we change columns name.

Reply:
> I'm sorry to reply you so late.

No problem! And thank you for the PR!

I think I understand the intention of the code, but `select foo from test where name = "*"` or even `update some_table set selected = 42 where name = "*"` would also reset the columns, and generally, it might not be problem to always reset the columns for every execution of the prepared statement (unless that would be a performance bottleneck for some drivers). That appears to be cleaner than trying to detect select statements that need it.

Reply:
Oh my god,  I did not take that into mind.

Reply:
@cmb69 Thanks for your review.
I should fix this from another perspective.